scripts/esbuild/mock-doc.ts

```diff
@@ -0,0 +1,73 @@
+import type { BuildOptions as ESBuildOptions } from 'esbuild';
+import fs from 'fs-extra';
+import { join } from 'path';
+
+import { writeSizzleBundle } from '../bundles/plugins/sizzle-plugin';
+import { bundleDts } from '../utils/bundle-dts';
+import { BuildOptions } from '../utils/options';
+import { writePkgJson } from '../utils/write-pkg-json';
+import { getBaseEsbuildOptions, getEsbuildAliases, runBuilds } from './util';
+
+export async function buildMockDoc(opts: BuildOptions) {
+  const inputDir = join(opts.buildDir, 'mock-doc');
+  const srcDir = join(opts.srcDir, 'mock-doc');
+  const outputDir = opts.output.mockDocDir;
+
+  // bundle d.ts
+  await bundleMockDocDts(opts, inputDir, outputDir);
+
+  writePkgJson(opts, outputDir, {
+    name: '@stencil/core/mock-doc',
+    description: 'Mock window, document and DOM outside of a browser environment.',
+    main: 'index.cjs',
+    module: 'index.js',
+    types: 'index.d.ts',
+    sideEffects: false,
+  });
+
+  const mockDocAliases = getEsbuildAliases();
+
+  const sizzlePath = await writeSizzleBundle(opts);
+  mockDocAliases['sizzle'] = sizzlePath;
+
+  const mockDocBuildOptions: ESBuildOptions = {
+    ...getBaseEsbuildOptions(),
+    entryPoints: [join(srcDir, 'index.ts')],
+    bundle: true,
+    alias: mockDocAliases,
+    logLevel: 'info',
+    target: 'node16',
+  };
+
+  const esmOptions: ESBuildOptions = {
+    ...mockDocBuildOptions,
+    format: 'esm',
+    outfile: join(outputDir, 'index.js'),
+  };
+
+  const cjsOptions: ESBuildOptions = {
+    ...mockDocBuildOptions,
+    format: 'cjs',
+    outfile: join(outputDir, 'index.cjs'),
```

Comment:
Follow up - it looks like [`banner` is also the acceptable way to add `strict`](https://github.com/evanw/esbuild/issues/422#issuecomment-739740602) - should we do that as well? [Here's where we use this option today](https://github.com/ionic-team/stencil/blob/9d2d0a0612caafb88c559907c6dec1a39e760615/scripts/bundles/mock-doc.ts#L43)

Reply:
yeah I think we'll need to put that in if we want it, I'll throw it in the banner (it looks like that was more or less what we were doing before anyway)

Reply:
actually I don't think this is necessary! it looks like because this `cjs` output is being built from ES Modules (our `.ts` files) it can automatically determine that it's safe to put the whole output file in strict mode. So if I `cat mock-doc/index.cjs | head -n 4` after building I get the following:

```js
/*!
 Stencil Mock Doc (CommonJS) v4.7.2-dev.1700164570.7d29ef4 | MIT Licensed | https://stenciljs.com
 */
"use strict";
```